File: snippets/syntax/SyntaxTest.py
# Sample Python file for testing VSCode theme syntax highlighting
import os
from typing import List, Dict, Optional
from dataclasses import dataclass
import asyncio

# Constants
MAX_COUNT: int = 100
GREETING: str = "Hello, Python!"

@dataclass
class SyntaxTest:
    """A sample class to demonstrate syntax highlighting."""
    name: str
    count: int = 0
    items: List[str] = None

    def __post_init__(self):
        self.items = []

    def perform_actions(self) -> None:
        # Local variables
        number: int = 42
        pi: float = 3.14159
        is_active: bool = True

        # If statement
        if number > 0:
            print(f"{GREETING} {self.name}")
        elif number == 0:
            print("Zero detected")
        else:
            print("Negative number")

        # List comprehension
        self.items = [f"Item {i}" for i in range(MAX_COUNT)]

        # Match statement (Python 3.10+)
        color = "blue"
        match color:
            case "red":
                print("Color is red")
            case "green":
                print("Color is green")
            case "blue":
                print("Color is blue")
            case _:
                print("Unknown color")

        # Lambda function
        printer = lambda s: print(f"Lambda: {s}")
        for item in self.items:
            printer(item)

        # Exception handling
        try:
            result = 10 / number
            print(f"Result: {result}")
        except ZeroDivisionError as e:
            print(f"Error: {e}")
        finally:
            print("Cleanup complete")

        # Dictionary
        scores: Dict[str, int] = {"Alice": 95, "Bob": 85}

    @staticmethod
    def static_method() -> str:
        return "Static method called"

# Decorator
def log_decorator(func):
    def wrapper(*args, **kwargs):
        print(f"Calling {func.__name__}")
        return func(*args, **kwargs)
    return wrapper

@log_decorator
def divide(a: int, b: int) -> Optional[float]:
    if b == 0:
        raise ValueError("Division by zero")
    return a / b

# Async function
async def async_example() -> None:
    print("Starting async task")
    await asyncio.sleep(1)
    print("Async task complete")

# Generator
def item_generator(max_items: int):
    for i in range(max_items):
        yield f"Generated item {i}"

# Main block
if __name__ == "__main__":
    test = SyntaxTest(name="TestUser")
    test.perform_actions()
    print(test.static_method())

    # Async execution
    asyncio.run(async_example())

    # Generator usage
    for item in item_generator(5):
        print(item)

    # F-string with expressions
    print(f"Sum: {1 + 2}, List: {[x for x in range(3)]}")

    # Type hints with complex types
    def process_data(data: List[Dict[str, int]]) -> None:
        print(f"Processing {data}")

    process_data([{"key": 1}, {"key": 2}])

    # Set and tuple
    unique_items = {1, 2, 2, 3}
    coordinates = (10, 20)
    print(f"Set: {unique_items}, Tuple: {coordinates}")

File: snippets/syntax/SyntaxTest.js
// Sample JavaScript file for testing VSCode theme syntax highlighting
import { createServer } from 'http';

// Constants
const MAX_COUNT = 100;
const GREETING = 'Hello, JavaScript!';

// Class
class SyntaxTest {
  #name; // Private field
  #count = 0;
  items = [];

  constructor(name) {
    this.#name = name;
  }

  // Method with various syntax elements
  performActions = async () => {
    // Local variables
    const number = 42;
    const pi = 3.14159;
    const isActive = true;

    // If statement
    if (number > 0) {
      console.log(`${GREETING} ${this.#name}`);
    } else if (number === 0) {
      console.log('Zero detected');
    } else {
      console.log('Negative number');
    }

    // For loop with destructuring
    for (let i = 0; i < MAX_COUNT; i++) {
      this.items.push(`Item ${i}`);
    }

    // Object destructuring
    const { name: userName } = { name: 'Alice', score: 95 };
    console.log(`User: ${userName}`);

    // Switch
    const color = 'blue';
    switch (color) {
      case 'red':
        console.log('Color is red');
        break;
      case 'green':
        console.log('Color is green');
        break;
      case 'blue':
        console.log('Color is blue');
        break;
      default:
        console.log('Unknown color');
    }

    // Arrow function
    const printer = (s) => console.log(`Arrow: ${s}`);
    this.items.forEach(printer);

    // Try-catch with async
    try {
      const result = await this.#divide(10, number);
      console.log(`Result: ${result}`);
    } catch (error) {
      console.error(`Error: ${error.message}`);
    }

    // Map and spread operator
    const scores = new Map([['Alice', 95], ['Bob', 85]]);
    const scoreArray = [...scores.entries()];
    console.log('Scores:', scoreArray);
  };

  // Private method
  #divide = async (a, b) => {
    if (b === 0) throw new Error('Division by zero');
    return a / b;
  };

  // Getter
  get name() {
    return this.#name;
  }
}

// IIFE (Immediately Invoked Function Expression)
(function () {
  console.log('IIFE executed');
})();

// Generator function
function* itemGenerator(maxItems) {
  for (let i = 0; i < maxItems; i++) {
    yield `Generated item ${i}`;
  }
}

// Async function
async function runServer() {
  const server = createServer((req, res) => {
    res.writeHead(200, { 'Content-Type': 'text/plain' });
    res.end('Hello, World!\n');
  });

  server.listen(3000, () => {
    console.log('Server running on port 3000');
  });
}

// Main execution
const main = async () => {
  const test = new SyntaxTest('TestUser');
  await test.performActions();

  // Generator usage
  for (const item of itemGenerator(5)) {
    console.log(item);
  }

  // Template literal with expressions
  console.log(`Sum: ${1 + 2}, Array: ${[1, 2, 3]}`);

  // Optional chaining and nullish coalescing
  const obj = { data: { value: 42 } };
  console.log(obj?.data?.value ?? 'Default');
};

// Execute main
main().catch(console.error);

// Export
export { SyntaxTest };